internal/service/iam/role_test.go

```diff
@@ -536,12 +592,18 @@ func TestAccIAMRole_permissionsBoundary(t *testing.T) {
 					"force_destroy",
 				},
 			},
+			// Test empty value
+			{
+				Config:      testAccRoleConfig_permissionsBoundary(rName, ""),
+				ExpectError: regexache.MustCompile(`Value "" cannot be parsed as an ARN.`),
+			},
```

Comment:
We can't change the acceptable values here until the next major version. `permissions_boundary` will have to accept an empty string and treat it as no value. Validation should only check for a valid ARN if it's not an empty string